pkg/build/path_rewriter.go

```diff
@@ -49,5 +46,23 @@ func (rw *RelativePathRewriter) MakeAbs(parent, obj *url.URL) *url.URL {
 	}
 }
 
+// Abs makes an absolute url for URL that has an empty or file-based
+// scheme.
+func (rw *RelativePathRewriter) Abs(obj *url.URL) (*url.URL, error) {
+	npath := obj.Path
+	if (obj.Scheme == "file" || obj.Scheme == "") && !filepath.IsAbs(obj.Path) {
```

Comment:
consider obj.Scheme = " dlfjkjasd;lkfjas"

This function returns nil error.

Reply:
That seems fine to me. This doesn't care about unknown schemes. But, I would think that URL parsing would fail earlier

Reply:
this function needs to work on its own, you shouldn't depend on other functions behavior.  just rewrite it so the unknown defaults to an error instead of nil error.

Reply:
Not following -- this makes it much cleaner. Here, Abs ignores all schemes it doesn't know about. I can add the checks here or in the parent file, but it's not clear to me which is better -- in the parent it adds duplication. 

I've changed the name to makeAbsForFileScheme to be clearer.